Detectors/FIT/FDD/simulation/include/FDDSimulation/Digitizer.h

```diff
@@ -32,49 +37,75 @@ class Digitizer
   typedef math_utils::RandomRing<float_v::size() * DigitizationParameters::mPheRRSize> HitRandomRingType;
   typedef math_utils::RandomRing<float_v::size() * DigitizationParameters::mHitRRSize> PheRandomRingType;
 
- public:
-  Digitizer(const DigitizationParameters& params, Int_t mode = 0) : mEventTime(0), mIntRecord(), mEventID(-1), mSrcID(-1), mMCLabels(), parameters(params), mTime(), mRndScintDelay(HitRandomRingType::RandomType::CustomTF1), mRndGainVar(PheRandomRingType::RandomType::CustomTF1), mRndSignalShape(PheRandomRingType::RandomType::CustomTF1), mPMResponseTables() { init(); };
-  ~Digitizer() = default;
+  using ChannelBCDataF = std::array<float, mNTimeBinsPerBC>;
 
-  void process(const std::vector<o2::fdd::Hit>* hits, o2::fdd::Digit* digit);
+ public:
+  struct BCCache : public o2::InteractionRecord {
+    std::array<ChannelBCDataF, mNchannels> pulse = {};
+    std::vector<o2::fdd::MCLabel> labels;
+
+    BCCache();
+
+    void clear()
+    {
+      for (auto& chan : pulse) {
+        chan.fill(0.);
+      }
+    }
+
+    BCCache& operator=(const o2::InteractionRecord& ir)
+    {
+      o2::InteractionRecord::operator=(ir);
+      return *this;
+    }
+    void print() const;
+  };
+
+  void process(const std::vector<o2::fdd::Hit>& hits,
+               std::vector<o2::fdd::Digit>& digitsBC,
+               std::vector<o2::fdd::ChannelData>& digitsCh,
+               o2::dataformats::MCTruthContainer<o2::fdd::MCLabel>& labels);
+
+  void flush(std::vector<o2::fdd::Digit>& digitsBC,
+             std::vector<o2::fdd::ChannelData>& digitsCh,
+             o2::dataformats::MCTruthContainer<o2::fdd::MCLabel>& labels);
 
-  void initParameters();
   void SetEventTime(long value) { mEventTime = value; }
   void SetEventID(Int_t id) { mEventID = id; }
   void SetSrcID(Int_t id) { mSrcID = id; }
-  void SetInteractionRecord(uint16_t bc, uint32_t orbit)
-  {
-    mIntRecord.bc = bc;
-    mIntRecord.orbit = orbit;
-  }
-  const o2::InteractionRecord& GetInteractionRecord() const { return mIntRecord; }
-  o2::InteractionRecord& GetInteractionRecord(o2::InteractionRecord& src) { return mIntRecord; }
-  void SetInteractionRecord(const o2::InteractionRecord& src) { mIntRecord = src; }
-  uint32_t GetOrbit() const { return mIntRecord.orbit; }
-  uint16_t GetBC() const { return mIntRecord.bc; }
+  void SetInteractionRecord(const o2::InteractionTimeRecord& src) { mIntRecord = src; }
 
   void SetTriggers(o2::fdd::Digit* digit);
   Int_t SimulateLightYield(Int_t pmt, Int_t nPhot);
-  Float_t SimulateTimeCFD(Int_t channel);
+  Float_t SimulateTimeCFD(ChannelBCDataF pulse);
```

Comment:
Provide cons reference rather than the copy of the pulse, i.e. ``SimulateTimeCFD(const ChannelBCDataF& pulse);``

Reply:
Done